### PR TITLE
[DROOLS-7130] Fix unresolved type exception inside kogito-apps/org.kie.kogito:jitexecutor-runner

### PR DESCRIPTION
**Pull request by gitgabrio**
@radtriste  @cristianonicolai  @tarilabs @danielezonca @pibizza @baldimir 

**JIRA**:
[link](https://issues.redhat.com/browse/DROOLS-7130)

These PR

1. declares reflect-config files for discovery during native image compilation
2. completely remove direct dependency between kie-dmn and kie-pmml-trusty
3. inside kie-dmn, feature the efesto framework to invoke pmml execution

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] tests</b>

- for a <b>full downstream build</b> 
  - for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

- <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

- <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] native</b>

- for <b>mandrel checks</b>  
  Run native checks against Mandrel image
  Please add comment: <b>Jenkins run mandrel</b>

- for a <b>specific mandrel check</b>  
  Run native checks against Mandrel image  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] mandrel</b>
</details>

### CODE REVIEW

**Comment by gitgabrio** [comment]
Jenkins run native

**Comment by baldimir** [comment]
Jenkins execute fdb

**Comment by baldimir** [comment]
Jenkins run fdb

File: kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/pmml/DMNKiePMMLTrustyInvocationEvaluator.java [hunk 1]
```diff
@@ -37,26 +38,23 @@
...
+import org.kie.efesto.runtimemanager.api.model.EfestoInput;
 import org.kie.efesto.runtimemanager.api.model.EfestoOutput;
+import org.kie.efesto.runtimemanager.api.model.EfestoRuntimeContext;
 import org.kie.efesto.runtimemanager.api.service.RuntimeManager;
-import org.kie.memorycompiler.KieMemoryCompiler;
-import org.kie.pmml.api.compilation.PMMLCompilationContext;
-import org.kie.pmml.api.runtime.PMMLRuntimeContext;
-import org.kie.pmml.compiler.PMMLCompilationContextImpl;
-import org.kie.pmml.evaluator.core.PMMLRuntimeContextImpl;
-import org.kie.pmml.evaluator.core.model.EfestoInputPMML;
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/pmml/DMNKiePMMLTrustyInvocationEvaluator.java:51]: I believe this the key change (EfestoInputPMML -> EfestoInput) which satisfies your preferred Efesto guideline and I have no objection to that, thanks for the refactor.

**Review by tarilabs — COMMENTED** [review]
The changes inside `/kie-dmn` looks okay to me, and if this globally suiting the Efesto guideline better in your view and you are more satisfied with it, all the better, thank you for the contribution!

> 2. completely remove direct dependency between kie-dmn and kie-pmml-trusty

Not sure what this is referring to:

- if this is referring to the situation after the merge of DROOLS-7037, I have no additional comment
- if this is referring to some prexisting situation, I believe we discussed this point at large during Monday and today meeting, and I've tried to motivate why this assertion is technically incorrect, likewise `scope=provided` is not "wrong" per se, as we used this paradigm for many years now eg. in Java EE or Jakarta.

This PR is one of the way with the intent to address a current NI compilation issue, good with me 👍

**Review by baldimir — CHANGES_REQUESTED** [review]
Thank you for fixing this. Based on the SonarCloud analysis, there are 2 new code smells and some new methods not covered by tests. Could you please fix those things from the analysis?

**Comment by gitgabrio** [comment]
Hi @tarilabs 
> * if this is referring to some prexisting situation, I believe we discussed this point at large during Monday and today meeting, and I've tried to motivate why this assertion is technically incorrect, likewise `scope=provided` is not "wrong" per se, as we used this paradigm for many years now eg. in Java EE or Jakarta.
> 
Just to (try to) better explain my point.
What I'm referring to are exactly the removed imports that comes from some kie-pmml-module. 
Without those modules declared as dependency, the class with those imports simply could not compile, and this is the direct dependency/binding I'm referring to.

File: kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/pmml/DMNKiePMMLTrustyInvocationEvaluator.java (hunk 1 above)
Comment (gitgabrio) [reply kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/pmml/DMNKiePMMLTrustyInvocationEvaluator.java:51]: :+1:

**Comment by cristianonicolai** [comment]
Jenkins run native

**Comment by cristianonicolai** [comment]
jenkins run quarkus-lts

**Review by tarilabs — APPROVED** [review]
Approval on the basis discussed in the meetings.

**Comment by gitgabrio** [comment]
@tarilabs  @baldimir @danielezonca @pibizza 
As per private discussion, since this is a blocker issue, let's ignore for the moment being the sonar analysys.
As far as I see, the unsucessfull build failed due to 
`[ERROR]  You have exceeded a secondary rate limit. Please wait a few minutes before you try again.`

**Comment by baldimir** [comment]
I am ok with ignoring Sonar for now, but please create a ticket to fix the problems reported in the analysis.

**Review by danielezonca — APPROVED** [review]

**Review by baldimir — APPROVED** [review]

**Comment by gitgabrio** [comment]
@kiegroup/gatekeepers

Could you please merge this, squashing commits? Thanks!

**Review by mariofusco — APPROVED** [review]

**Comment by radtriste** [comment]
@gitgabrio @mariofusco Could we cherry-pick this one to 8.27.x ?

EDIT: 1.27.x is also broken. See https://kie.zulipchat.com/#narrow/stream/236603-kogito-ci/topic/.5B1.2E27.2Ex.5D.20Kogito.20Apps.20-.20Native.20check

**Comment by mariofusco** [comment]
@radtriste drools 8.27.0.Beta is already out and I don't think we will create another patch release from it. Can you please explain what's the point in doing this cherry-pick? Which downstream project will use it?

**Comment by radtriste** [comment]
It is the productized branch. That is why I was suggesting.

**Comment by tarilabs** [comment]
> It is the productized branch. That is why I was suggesting.

@radtriste @cristianonicolai for my understanding, but is the `kogito-apps/jitexecutor` part of the productized branch?
This PR main goal is to solve a GraalVM NI compilation induced issue specifically in `kogito-apps/jitexecutor`.

**Comment by cristianonicolai** [comment]
@tarilabs you're correct , this is not exactly needed for the product. The issue is that we do pipelines in the branch that is productized, but without filtering down to only productized profile. We can get around that, I'll just disable native build for `jitexecutor` in `1.27.x`.

**Pull request merged**